.golangci.yml

```diff
@@ -0,0 +1,67 @@
+# options for analysis running
+run:
+  # default concurrency is a available CPU number
+  concurrency: 4
+
+  # timeout for analysis, e.g. 30s, 5m, default is 1m
+  timeout: 5m
+linters:
+  # please, do not use `enable-all`: it's deprecated and will be removed soon.
+  # inverted configuration with `enable-all` and `disable` is not scalable during updates of golangci-lint
+  disable-all: true
+  enable:
+    - bodyclose
+    - deadcode
+    - depguard
+    - dogsled
+    - dupl
+    - errcheck
+    - funlen
+    - goconst
+    - gocritic
+    - gocyclo
+    - gofmt
+    - goimports
+    - golint
+    - goprintffuncname
+    - gosimple
+    - govet
+    - ineffassign
+    - interfacer
+    - misspell
+    - nolintlint
+    - rowserrcheck
+    - scopelint
+    - staticcheck
+    - structcheck
+    - stylecheck
+    - typecheck
+    - unconvert
+    - unparam
+    - unused
+    - varcheck
+    - whitespace
```

Comment:
If there will be too many issues to solve we may limit the scope of this PR to some set of those rules and create Github issues asking for introducing additional checks. It this way we may create multiple easy good first issues 😉

Reply:
Good Idea. 

I can reduce some linters based on the value of this pr, and make the current pr function as single as possible.

Reply:
I have disabled most of the linter, so that the current code can be passed by ci, and then we can resolve to enable the new linter by submitting an issue and pr

Reply:
In case of golint we will etiher have to disable one check or make breaking change: https://github.com/kedacore/keda/pull/1088#discussion_r484253257

Reply:
For golint, we only need to modify the struct field restApiTemplate in `pkg/scalers/artemis_scaler.go:33:2`, without changing metadata["restApiTemplate"] to metadata["restAPITemplate"], so this change will not break the scaler .
```
➜ keda git:(add-golangci) ✗ golint ./...
pkg/scalers/artemis_scaler.go:33:2: struct field restApiTemplate should be restAPITemplat
```

In this pr, I disabled golint, and I plan to modify this in the new pr if the pr is merged.

Reply:
That's interesting but sounds good to me